project_tracker_template/update_dates_priority_checkboxes.gs

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Updates start date, estimated launch date, priority of a task/ milestone in
+ * the Task sheet
+ * Updates checkboxes reflecting milestone assignment in the Team Sheet
+ * @param {Date} projectStart Date object containing start date of the project
+ * @param {Date} currDate Date object containing current date of the project
+ * @param {Boolean} isProjectStartAfterToday Boolean value to store whether
+ *     project start date is after current date (TRUE) or not (FALSE).
+ * @param {Number[]} codingUnitsPerDay Array of numbers containing coding units
+ *     per day of each engineer
+ * @param {Object[][]} engineerInfo A two dimensional array containing
+ *     information of the engineers in the Team Sheet
+ * @param {String[]} usernames Array of strings containing username of each
+ *     engineer
+ * @param {Object[][]} taskTable A two dimensional array containing values of
+ *     the table in the Task Sheet
+ * @param {Number} tableLength Length of taskTable array
+ * @param {Number} countEng Number of engineers in the Team
+ * @param {Object[][]} checkBoxValues A two dimensional array to reset checkbox
+ *     selections in the Team Sheet
+ */
+function updateDatesPriorityCheckbox(projectStart, currDate,
+                                     isProjectStartAfterToday, engineerInfo,
+                                     codingUnitsPerDay, usernames, taskTable,
+                                     countEng, tableLength, checkBoxValues) {
+  // Array containing task start dates to display in the start date column of
+  // Task Sheet
+  var displayStartDateArray =
+      getDisplayStartDate(projectStart, engineerInfo, countEng);
+  // Boolean array storing whether task start date (to display) is after current
+  // date
+  var isStartDateAfterTodayArray = [];
+  // Array containing task start dates to calculate estimated launch dates in
+  // the launch date column of Task Sheet
+  var actualStartDateArray = [];
+  // Initialize arrays 'actualStartDateArray' and 'isStartDateAfterTodayArray'
+  getActualStartDate(displayStartDateArray, isStartDateAfterTodayArray,
+                     actualStartDateArray, countEng, currDate);
+  const serialNumberColumn = 0;
```

Comment:
const should be declared with all caps like SERIAL_NUMBER_COLUMN

Also I think all const should be declared in the global scope so that they don't have to be re-initialized on every call